File: src/main/java/pl/mjedynak/idea/plugins/builder/factory/PackageChooserDialogFactory.java
package pl.mjedynak.idea.plugins.builder.factory;

import com.intellij.ide.util.PackageChooserDialog;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

public class PackageChooserDialogFactory {

    private PackageChooserDialogFactory() {
        throw new UnsupportedOperationException("Utility class");
    }

    @Contract("_, _ -> new")
    public static @NotNull PackageChooserDialog getPackageChooserDialog(
            @NotNull String message, @NotNull Project project) {
        return new PackageChooserDialog(message, project);
    }
}

File: src/main/kotlin/io/github/kingg22/kobra/builder/factory/PackageChooserDialogFactory.kt
package io.github.kingg22.kobra.builder.factory

import com.intellij.ide.util.PackageChooserDialog
import com.intellij.openapi.project.Project
import org.jetbrains.annotations.Contract

public object PackageChooserDialogFactory {
    @JvmStatic
    @Contract("_, _ -> new")
    public fun getPackageChooserDialog(message: String, project: Project): PackageChooserDialog =
        PackageChooserDialog(message, project)
}
